views/js/qtiXmlRenderer/tpl/responseDeclaration.tpl

```diff
@@ -11,7 +11,11 @@
     {{~#if correctResponse.length}}
     <correctResponse>
         {{~#each correctResponse}}
+        {{~#if ../isRecord}}
+        <value fieldIdentifier="{{fieldIdentifier}}" baseType="{{baseType}}"><![CDATA[{{{value}}}]]></value>
```

Comment:
A Record may be recursive. So I believe we need to manage the case of having a recursive object.